src/aggregator/aggregator/forwarded_writer.go

```diff
@@ -230,12 +235,9 @@ func (w *forwardedWriter) Flush() error {
 // NB: Do not close the client here as it is shared by all the forward
 // writers. The aggregator is responsible for closing the client.
 func (w *forwardedWriter) Close() error {
-	if w.closed {
+	if w.closed.Swap(true) {
```

Comment:
`Swap` returns previous value. If `Close()` is called more than once, this would cause 2nd and subsequent calls return error which it probably shouldn't because this function doesn't do anything besides setting the `w.closed`.

Should we use `w.closed.Store(true)` instead?

Reply:
it might still be useful for tests, though - and is a pervasive pattern around here.